salesagility/SuiteCRM
# Fixed #3645 - User defined smtp settings not used when sending email

Added inbound_email_id when sending mail in order to get correct smtp settings

## Description
fix #3645 

## Motivation and Context
Changes needed to send from user specified smtp settings

## How To Test This
1. You need a way to know which SMTP server is used when sending. An easy way is to use gmail smtp because it will always set the from_address to the smtp user; make sure you use one gmail or google apps account for the system smtp 
2. Under a test user, create a new entry under Outgoing SMTP Mail Servers under Profile - scroll all the way down - Settings - Mail Accounts; here make sure to use another gmail or google apps account so that it will be clear whether that smtp user is used or not when receiving email.
2. Create or edit the corresponding incoming account under Mail account, select SMTP settings created previously. Save all and close.
3. Compose new email and send to an email address you can access. Check email and confirm it is from user smtp userid, not from system smtp userid.

## Types of changes
- [ x] Bug fix (non-breaking change which fixes an issue)

### Final checklist
- [x ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.
